Management/src/ActuatorApi/BasicAuthExtensions.cs

```diff
@@ -0,0 +1,41 @@
+using System.Security.Claims;
+using idunno.Authentication.Basic;
+using Microsoft.AspNetCore.Authentication;
+
+namespace Steeltoe.Samples.ActuatorApi;
+
+internal static class BasicAuthExtensions
+{
+    public static void ConfigureActuatorAuth(this IServiceCollection services)
+    {
+        AuthenticationBuilder builder = services.AddAuthentication(BasicAuthenticationDefaults.AuthenticationScheme);
+
+        builder.AddBasic(BasicAuthenticationDefaults.AuthenticationScheme, options =>
+        {
+            // AllowInsecureProtocol = true is not secure. Do not use in a real application.
+            options.AllowInsecureProtocol = true;
```

Comment:
This was added recently. What do we need this for?

Reply:
The basic auth library is supposed to throw HTTP 421 when you try to use over HTTP without this. I think I only got the 421 in place of 401 though - I'm not clear why it seems to have successfully authorized requests without this, but maybe I had been primarily testing against the https profile?? regardless, this is necessary to get accurate response codes when using HTTP

Reply:
Then you're a horrible person :)

![image](https://github.com/user-attachments/assets/96ab3082-3f76-4d35-900a-ac73472e34eb)

It is kinda odd that we're trying to offer a secure-by-default sample using basic auth, while that's fundamentally insecure over HTTP. But we can't use https due to certificate trust issues. Do I understand this correctly?

Reply:
I do what I can 😄.  Yeah, it's less than ideal, but it's probably the simplest solution that shows the _potential_ and this kind of security isn't that important for local dev